misc/drone-scripts/validate-and-serialize.sh

```diff
@@ -13,33 +13,101 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# datastore name refers storage backed by base ESX (physical ESX) where
+# ongoing builds are going to create file which helps to serialize CI run
 DS='-ds=datastore1'
-unit_test_array=($TEST_URL_ARRAY)
-numServers=${#unit_test_array[@]}
+
+# Lock variables to be consumed
+# build_lock:
+#       Overall lock makes sure only one thread at a time runs
+# check_lock:
+#       Represents a lock to perform cleanup leftover from previous run builds.
+build_lock="build_lock"
+check_lock="check_lock"
+
+# refers current CI build#
 DRONE_BUILD_NUMBER=${DRONE_BUILD_NUMBER:=0}
-prevBuildStatus=`drone build info vmware/docker-volume-vsphere $(( $DRONE_BUILD_NUMBER-$numServers ))`
-outArray=($prevBuildStatus)
+echo "DRONE_BUILD_NUMBER = $DRONE_BUILD_NUMBER"
 
-# Check if any other build is in the ongoing folder, if present, check if entry is stale.
-# If entry is stale clean it up.
-# If entry is an on going build wait for it.
+# wait time in seconds
+interval=30
 
-prevBuild=`govc datastore.ls $DS docker-volume-vsphere/ongoing|tail -n 1`
+# Decide whether cleanup is needed or not.
+# grab ongoing build# and parse build info, returns true when build_status
+# is "*not* Running" or the request is from the same build number
+function is_cleanup_needed {
+    # Retrieve on-going build information
+    ongoingBuildInfo=`drone build info vmware/docker-volume-vsphere $1`
+    buildInfoObj=($ongoingBuildInfo)
 
-while [[ "$prevBuild" != "" ]];
-do
-    prevBuildStatus=`drone build info vmware/docker-volume-vsphere $prevBuild`
-    outArray=($prevBuildStatus)
-    if [[ ${outArray[2]} != *"running"* || ${outArray[2]} == *"running"* && $prevBuild == $DRONE_BUILD_NUMBER ]]
+    # verifies status value OR request is resulted from *Restart* event
+    if [[ ${buildInfoObj[2]} != *"running"* || $1 == $DRONE_BUILD_NUMBER ]]
     then
-        govc datastore.rm $DS docker-volume-vsphere/ongoing/$prevBuild
+        return 0
     else
-        echo "Waiting 2 minutes for previous build $prevBuild to complete";
-        sleep 120;
+        return 1
+    fi
+}
+
+# Util to remove created directory (consumed as locking mechanism or
+# preserving metadata for ongoing build)
+function release_lock {
+#    echo "Releasing lock file [$1]"
```

Comment:
either drop $1 check in acquire_lock, or add it here - acquire_lock and release_lock should handle the args the same way